torchrl/envs/vec_env.py

```diff
@@ -1004,6 +1002,10 @@ def _run_worker_pipe_shared_mem(
                 raise RuntimeError("call 'init' before resetting")
             # _td = tensordict.select("observation").to(env.device).clone()
             _td = env._reset(**reset_kwargs)
+            _td.set_default(
+                "done",
+                torch.zeros(*_td.batch_size, 1, dtype=torch.bool, device=_td.device),
```

Comment:
one issue with set_default is that we're creating the tensor at every single call, thereby allocating memory that is then collected if unused. 
Depending on the kind of env you're running this may or may not cause some overhead. In practice, a boolean tensor is quite cheap so I might be overthinking it.
The solutions might be: 
1. leave it as it is
2. use `torch.zeros(1, ...).expand(*_td.batch_size, 1)` which uses less memory but requires more operations
3. (related to 1) implement a `setdefault` with callable, though this has been discussed by the python community and they [decided not to implement it](https://bugs.python.org/issue31240) for some valid reasons. Since we'd like to stay as close as we can from dict I would not vote for it (besides, we have a `set_default` method when it should be `setdefault` 😱)
4. do a `try / except KeyError` but the issue here is that `try / except` causes an overhead of its own when the except is reached (which should happen less often than not, thereby reversing the memory overhead that occurs say 90%. of the time to a compute time overhead that occurs 10% of the time)
5. check if the key is there, and if not populate. This is basically what set_default does so I would not expect this to be slower, but obviously less 'atomic'.  

Also tensordict do not always have a device (ie device can be None). This is to enable tensordict that collect all the state_dict of a module distributed across several GPUs, for instance. 
Not sure it is relevant here as the tensordict will have the device of the env and the env must have a device, so it's unlikely that it will cause any bug any time soon.

Reply:
I leave the choice to you. Whatever we choose we should also update the normeal `EnvBase.resert()` then, as this is what is done there https://github.com/pytorch/rl/blob/main/torchrl/envs/common.py#L454

My personal opinion is that all this checks on the done flag should be removed